### PR TITLE
Document cudaMallocAsync

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: docs/gpu_memory_allocation.rst [hunk 1]
```diff
@@ -70,3 +70,30 @@ Common causes of OOM failures
   memory. Note however, that the algorithm is basic and you can often get better 
   trade-off between compute and memory by disabling the automatic remat pass and doing
   it manually with `the jax.remat API <https://jax.readthedocs.io/en/latest/jep/11830-new-remat-checkpoint.html>`_
+
+
+Experimental feature
```
Comment (dfm) [inline docs/gpu_memory_allocation.rst:75]: Nit:

```suggestion
Experimental features
```

Perhaps?

File: docs/gpu_memory_allocation.rst [hunk 2]
```diff
@@ -70,3 +70,30 @@ Common causes of OOM failures
...
+  This will remove the big fixed pre-allocation and use a memory pool that grows.
+  The expected benefit is no need to set `XLA_PYTHON_CLIENT_MEM_FRACTION`.
+
+  The risk are:
+  - that memory fragmentation is different, so if you are close to the
+    limit, the exact OOM case due to fragmentation will be different.
+  - The allocation time won't be all paid at the start, but be incurred
+    when the memory pool need to be increased. So you could
+    experience less speed stability at the start and for benchmarks
+    it will be even more important to ignore the first few iterations.
```
Comment (dfm) [inline docs/gpu_memory_allocation.rst:92]: It looks like the CI failures are real and associated with this block here. My best guess is that lists aren't allowed within a block quote in RST like this. What about rewording this into a single paragraph?

**Review by dfm — CHANGES_REQUESTED** [review]
Small comments inline, but otherwise looks good. Thanks!

File: docs/gpu_memory_allocation.rst (hunk 2 above)
Comment (nouiz) [reply docs/gpu_memory_allocation.rst]: I fixed it with a new line. The CI documentation now passes.

**Comment by dfm** [comment]
Thanks! Can you squash commits then we can merge?

**Comment by nouiz** [comment]
> Thanks! Can you squash commits then we can merge?

Done

**Review by dfm — APPROVED** [review]

**Pull request merged**